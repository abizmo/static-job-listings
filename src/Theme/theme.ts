const theme = {
  colors: {
    100: 'hsl(180, 52%, 96%)',
    200: 'hsl(180, 31%, 95%)',
    500: 'hsl(180, 29%,50%)',
    600: 'hsl(180, 8%, 52%)',
    800: 'hsl(180, 14%, 20%)',
    black: 'hsl(180, 0%, 0%)',
    white: 'hsl(180, 100%, 100%)',
  },
  fontSizes: {
    200: '.6875rem',
    400: '.875rem',
    500: '.9375rem',
    600: '1rem',
    700: '1.125rem',
    900: '1.375rem',
  },
  fontFamily: "'League Spartan', sans-serif",
  sizes: {
    4: '.25rem',
    8: '.5rem',
    12: '.75rem',
    16: '1rem',
    20: '1.25rem',
    24: '1.5rem',
    32: '2rem',
    48: '3rem',
    88: '5.5rem',
  },
};

export default theme;
