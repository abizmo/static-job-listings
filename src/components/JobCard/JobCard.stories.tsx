import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import JobCard from './JobCard';

export default {
  title: 'Components/JobCard',
  component: JobCard,
} as ComponentMeta<typeof JobCard>;

const Template: ComponentStory<typeof JobCard> = (args) => (
  <JobCard {...args} />
);

export const Default = Template.bind({});
Default.args = {
  id: 1,
  company: 'Photosnap',
  logo: './images/photosnap.svg',
  new: true,
  featured: true,
  position: 'Senior Frontend Developer',
  role: 'Frontend',
  level: 'Senior',
  postedAt: '1d ago',
  contract: 'Full Time',
  location: 'USA Only',
  languages: ['HTML', 'CSS', 'JavaScript'],
  tools: [],
};
