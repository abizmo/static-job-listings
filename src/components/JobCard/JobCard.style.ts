import styled from 'styled-components';

export const Title = styled.h3`
  color: ${({ theme }) => theme.colors[800]};
  font-size: ${({ theme }) => theme.fontSizes[900]};
  &:hover {
    color: ${({ theme }) => theme.colors[500]};
    cursor: pointer;
  }
`;

export const Header = styled.div`
  align-items: center;
  display: flex;
  gap: 24px;
  & div + div {
    margin-left: 8px;
  }
`;

export const Company = styled.h2(({ theme }) => ({
  fontSize: theme.fontSizes[500],
  '@media (min-width: 50rem)': {
    fontSize: theme.fontSizes[700],
  },
}));

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  gap: ${({ theme }) => theme.sizes[16]};
  list-style: none;
  &.bullet li {
    color: ${({ theme }) => theme.colors[600]};
    font-size: ${({ theme }) => theme.fontSizes[600]};
    font-weight: 500;
    @media (min-width: 50rem) {
      font-size: ${({ theme }) => theme.fontSizes[700]};
    }
  }
  &.bullet li + li {
    position: relative;
  }
  &.bullet li + li::before {
    background-color: ${({ theme }) => theme.colors[600]};
    border-radius: 50%;
    content: '';
    height: 4px;
    left: -10px;
    top: 50%;
    position: absolute;
    transform: translateY(-100%);
    width: 4px;
  }
`;

export const Wrapper = styled.div(({ theme }) => ({
  backgroundColor: theme.colors.white,
  borderRadius: theme.sizes[4],
  boxShadow: theme.boxShadow,
  display: 'flex',
  flexDirection: 'column',
  gap: theme.sizes[16],
  padding: `${theme.sizes[32]} ${theme.sizes[24]} ${theme.sizes[24]}`,
  position: 'relative',
  overflow: 'hidden',
  '&>div': {
    display: 'flex',
    flexDirection: 'column',
    gap: theme.sizes[16],
  },
  '&>ul': {
    borderTop: `1px solid ${theme.colors[600]}`,
    paddingTop: '1rem',
  },
  '&.feat::before': {
    backgroundColor: theme.colors[500],
    content: '""',
    height: '100%',
    left: 0,
    position: 'absolute',
    top: 0,
    width: '5px',
  },
  '@media(min-width: 50rem)': {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    '&>ul': {
      borderTop: 'none',
      paddingTop: 0,
    },
  },
}));
