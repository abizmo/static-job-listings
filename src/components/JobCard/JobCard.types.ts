type roles = 'Frontend' | 'Fullstack' | 'Backend';
type levels = 'Junior' | 'Senior' | 'Midweight';
type contracts = 'Full Time' | 'Part Time';

export interface Job {
  id: number;
  company: string;
  logo: string;
  new: boolean;
  featured: boolean;
  position: string;
  role: roles;
  level: levels;
  postedAt: string;
  contract: contracts;
  location: string;
  languages: string[];
  tools: string[];
  onFilter: (x: string) => void;
}
