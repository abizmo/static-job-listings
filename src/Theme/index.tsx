import React from 'react';
import { createGlobalStyle, css, ThemeProvider } from 'styled-components';

import theme from './theme';

const resetCss = css`
  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
  body,
  h1,
  p {
    margin: 0;
    padding: 0;
  }
  body,
  button {
    font-family: ${theme.fontFamily};
    font-weight: 700;
    line-height: 1.1;
  }
  button + button {
    margin-left: 0;
  }
`;

const GlobalStyle = createGlobalStyle`
  ${resetCss}
  body {
    background-color: ${theme.colors[100]};
    color: ${theme.colors[500]};
    height: 100vh;
  }
  #root:only-child {
    display: grid;
    grid-template-rows: 1fr min-content;
    height: 100vh;
  }
  main {
    place-self: center;
  }
`;

function Theme({ children }: any): JSX.Element {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
}

export default Theme;
