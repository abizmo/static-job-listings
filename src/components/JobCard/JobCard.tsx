import React from 'react';
import Avatar from '../Avatar/Avatar';
import Badge from '../Badge/Badge';
import Tag from '../Tag/Tag';
import {
  CardInfo,
  Company,
  Header,
  List,
  Title,
  Wrapper,
} from './JobCard.style';

import { Job } from './JobCard.types';

function JobCard(job: Job) {
  return (
    <Wrapper className={job.featured ? 'feat' : ''}>
      <CardInfo>
        <Avatar alt={job.company} src={job.logo} />
        <div>
          <Header>
            <Company>{job.company}</Company>
            <div>
              {job.new && <Badge type='new' />}
              {job.featured && <Badge type='feat' />}
            </div>
          </Header>
          <Title>{job.position}</Title>
          <List className='bullet'>
            <li>{job.postedAt}</li>
            <li>{job.contract}</li>
            <li>{job.location}</li>
          </List>
        </div>
      </CardInfo>
      <List>
        <li>
          <Tag category={job.role} onClick={() => {}} />
        </li>
        <li>
          <Tag category={job.level} onClick={() => {}} />
        </li>
        {job.languages.map((lang) => (
          <li key={lang}>
            <Tag category={lang} onClick={() => {}} />
          </li>
        ))}
        {job.tools.map((tool) => (
          <li key={tool}>
            <Tag category={tool} onClick={() => {}} />
          </li>
        ))}
      </List>
    </Wrapper>
  );
}

export default JobCard;
