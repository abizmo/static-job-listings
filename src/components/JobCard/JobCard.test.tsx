import { render, screen } from '@testing-library/react';
import React from 'react';

import data from '../../fixtures/data.json';
import Theme from '../../Theme';
import JobCard from './JobCard';
import { Job } from './JobCard.types';

const jobs = data as Job[];

describe('JobCard tset', () => {
  test('should render and match snapshot', () => {
    const { asFragment } = render(
      <Theme>
        <JobCard {...jobs[0]} />
      </Theme>
    );

    expect(screen.getByText(jobs[0].company)).toBeInTheDocument();
    expect(screen.getByText(jobs[0].position)).toBeInTheDocument();
    expect(screen.getByText(jobs[0].postedAt)).toBeInTheDocument();
    expect(screen.getByText(jobs[0].contract)).toBeInTheDocument();
    expect(screen.getByText(jobs[0].location)).toBeInTheDocument();
    expect(screen.getByText(jobs[0].languages[0])).toBeInTheDocument();
    expect(screen.getByText(/new/i)).toBeInTheDocument();
    expect(screen.getByText(/featured/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
