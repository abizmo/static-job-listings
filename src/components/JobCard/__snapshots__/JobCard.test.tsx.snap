// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard tset should render and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-ftvSup iyKVse feat"
  >
    <div
      class="sc-iBkjds jMHGKi"
    >
      <img
        alt="Photosnap"
        class="sc-bczRLJ iQWCWt"
        src="./images/photosnap.svg"
      />
      <div>
        <div
          class="sc-eCYdqJ jKdPYI"
        >
          <h2
            class="sc-jSMfEi gtlYBX"
          >
            Photosnap
          </h2>
          <div>
            <div
              class="sc-gsnTZi dHCAfR"
              type="new"
            >
              New!
            </div>
            <div
              class="sc-gsnTZi elznlA"
              type="feat"
            >
              Featured
            </div>
          </div>
        </div>
        <h3
          class="sc-hKMtZM fBxJUe"
        >
          Senior Frontend Developer
        </h3>
        <ul
          class="sc-gKXOVf fZpewS bullet"
        >
          <li>
            1d ago
          </li>
          <li>
            Full Time
          </li>
          <li>
            USA Only
          </li>
        </ul>
      </div>
    </div>
    <ul
      class="sc-gKXOVf fZpewS"
    >
      <li>
        <button
          class="sc-dkzDqf iShTWr"
          type="button"
        >
          <span>
            Frontend
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dkzDqf iShTWr"
          type="button"
        >
          <span>
            Senior
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dkzDqf iShTWr"
          type="button"
        >
          <span>
            HTML
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dkzDqf iShTWr"
          type="button"
        >
          <span>
            CSS
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dkzDqf iShTWr"
          type="button"
        >
          <span>
            JavaScript
          </span>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
